src/yaml-language-support/schema-utils.ts

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+import { KubernetesGroupKindNode, KubernetesResourceObjectBase } from "./schema-models";
+
+/**
+ * An utility function for splitting text into lines.
+ */
+export function splitLines(text: string): string[] {
+    return text.split(/\r\n?|\n/);
+} 
+
+/**
+ * A very easy utility function for decides whether the text is support to be a kubernetes yaml file, further
+ * implementations is needed to be more precise on the result rather than the regex test.
+ */
+export function isKubernetesYamlFile(text: string): boolean {
+    return !!(/\bkind\s*:/g.exec(text) ||
+        /\bapiVersion\s*:/g.exec(text));
+}
+
+/**
+ * An utility function for get apiVersion and kind from 'x-kubernetes-group-version-kind' node in kubernetes schema.
+ */
+export function parseApiVersionKind(groupKindNode: KubernetesGroupKindNode): KubernetesResourceObjectBase {
```

Comment:
The naming seems a bit odd here - returning `KubernetesResourceObjectBase` makes me think it's returning the resource object through a polymorphic interface - but all it's actually doing is returning the apiVersion/kind record.  So it's not clear to me whether `KubernetesResourceObjectBase` is meant to be a base type for k8s objects (in which case this function is returning the wrong thing), or a record type for apiVersion/kind (in which case its name is wrong).  Perhaps this will be clearer once we see it being used but it's hard to know looking at this code in isolation.